crates/rattler_upload/src/utils/server_util.rs

```diff
@@ -0,0 +1,250 @@
+use regex::Regex;
+use std::sync::OnceLock;
+use url::Url;
+
+/// Simplified server type without options
+#[derive(Debug, Clone, PartialEq)]
+pub enum SimpleServerType {
+    Quetz,
+    Artifactory,
+    Prefix,
+    Anaconda,
+    #[cfg(feature = "s3")]
+    S3,
+    CondaForge,
+    Unknown,
+}
+
+/// Server patterns
+struct ServerPatterns {
+    prefix: Regex,
+    anaconda: Regex,
+    quetz: Regex,
+    artifactory: Regex,
+    #[cfg(feature = "s3")]
+    s3: Regex,
+    conda_forge: Regex,
+}
+
+impl ServerPatterns {
+    fn new() -> Self {
+        Self {
+            // Prefix.dev patterns
```

Comment:
I wonder if using `regex` is the best choice. We could probably also do this with regular URL parsing and some `contains` queries and similar, e.g.:

```
url.host_str.contains(prefix.dev)
url.path().ends_with(/conda-forge)
url.protocol() == "s3" ... 
```